### PR TITLE
docs(user): add query strings tutorial

### LINKED ISSUE #475: Fill in the "Query Strings" section of tutorial.rst
Right now it just says "coming soon".

### PR DESCRIPTION
**Pull request by bssyousefi**
# Summary of Changes

Fill in the placeholder for `Query Strings` in the documentation. The placeholder only existed on **WSGI** tutorial, so the proposed change is just about that part of the documentation.

# Related Issues

Fixes #475.

# Pull Request Checklist

This is just a reminder about the most common mistakes.  Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!

If an item doesn't apply to your pull request, **check it anyway** to make it apparent that there's nothing to do.

- [ ] Applied changes to both WSGI and ASGI code paths and interfaces (where applicable).
- [ ] Added **tests** for changed code.
- [ ] Prefixed code comments with GitHub nick and an appropriate prefix.
- [x] Coding style is consistent with the rest of the framework.
- [x] Updated **documentation** for changed code.
    - [ ] Added docstrings for any new classes, functions, or modules.
    - [ ] Updated docstrings for any modifications to existing code.
    - [x] Updated both WSGI and ASGI docs (where applicable).
    - [ ] Added references to new classes, functions, or modules to the relevant RST file under `docs/`.
    - [ ] Updated all relevant supporting documentation files under `docs/`.
    - [ ] A copyright notice is included at the top of any new modules (using your own name or the name of your organization).
    - [ ] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/usage/restructuredtext/directives.html?highlight=versionadded#directive-versionadded).
- [ ] Changes (and possible deprecations) have [towncrier](https://towncrier.readthedocs.io/en/latest/quickstart.html#creating-news-fragments) news fragments under `docs/_newsfragments/`, with the file name format `{issue_number}.{fragment_type}.rst`. (Run `towncrier --draft` to ensure it renders correctly.)

If you have *any* questions to *any* of the points above, just **submit and ask**! This checklist is here to *help* you, not to deter you from contributing!

*PR template inspired by the attrs project.*

### CODE REVIEW

File: docs/user/tutorial.rst
Comment (CaselIT) [inline docs/user/tutorial.rst:1324]:
```suggestion
                    {'href': '/images/' + image} for image in images
```

File: docs/user/tutorial.rst
Comment (CaselIT) [inline docs/user/tutorial.rst:1319]: This version should be a static one like the above that serializes to msgpack, since we refer to it as "static"

File: docs/user/tutorial.rst
Comment (CaselIT) [inline docs/user/tutorial.rst:1379]:
```suggestion
                    {'href': '/images/' + image} for image in images
```

File: docs/user/tutorial.rst
Comment (CaselIT) [inline docs/user/tutorial.rst:1451]: maybe we could use -1 for the "defaut" here?

File: docs/user/tutorial.rst
Comment (CaselIT) [inline docs/user/tutorial.rst:1452]: since these are not really images, we could use `"images"` here and maybe say something like
```
Execute the following commands in order to simulate the creation of 3 files as images with different sizes. While these are not valid PNG files, they will work for this tutorial.
```

File: docs/user/tutorial.rst
Comment (CaselIT) [inline docs/user/tutorial.rst:1375]: I think it would be useful to showcase `get_param_as_int` here, so we could do
```suggestion
            max_size = int(req.get_param_as_int("maxsize", default=0, min_value=1))
```
(I've added a min value but we may omit it too)

**Review by CaselIT — CHANGES_REQUESTED** [review]
Thanks for the PR, great work!

I've left a few suggestions

**Comment by bssyousefi** [comment]
I have a question regarding a part of the code.
```
 _IMAGE_NAME_PATTERN = re.compile(
            '[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\.[a-z]{2,4}$'
        )
```
which I used from previous parts of the docs.
the string used as the pattern will raise error in python. I guess we should use one of the following ways
```
 _IMAGE_NAME_PATTERN = re.compile(
            '[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}.[a-z]{2,4}$'
        )
 _IMAGE_NAME_PATTERN = re.compile(
            r'[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\.[a-z]{2,4}$'
        )
```
However, I didn't try to fix it here and I thought that maybe I should first report it as an issue and then resolve it in another PR. Am I right or should I try the fix on this PR since it's a small change?

File: docs/user/tutorial.rst
Comment (vytas7) [inline docs/user/tutorial.rst:1309]: Could you shorten Rst text lines to a reasonable length, preferably around 80 characters?

File: docs/user/tutorial.rst
Comment (vytas7) [inline docs/user/tutorial.rst:1368]: Don't we need to update any example files on disk? And write tests for them too?

(See also: #2247.)

**Review by vytas7 — CHANGES_REQUESTED** [review]
Thanks for this contribution, looks good for the most of it.

We just need to clean up things a bit before merging (see my comments inline, as well as @CaselIT's remarks).

File: docs/user/tutorial.rst
Comment (bssyousefi) [reply docs/user/tutorial.rst]: Done

**Comment by bssyousefi** [comment]
> I have a question regarding a part of the code.
> 
> ```
>  _IMAGE_NAME_PATTERN = re.compile(
>             '[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\.[a-z]{2,4}$'
>         )
> ```
> 
> which I used from previous parts of the docs. the string used as the pattern will raise error in python. I guess we should use one of the following ways
> 
> ```
>  _IMAGE_NAME_PATTERN = re.compile(
>             '[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}.[a-z]{2,4}$'
>         )
>  _IMAGE_NAME_PATTERN = re.compile(
>             r'[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\.[a-z]{2,4}$'
>         )
> ```
> 
> However, I didn't try to fix it here and I thought that maybe I should first report it as an issue and then resolve it in another PR. Am I right or should I try the fix on this PR since it's a small change?

@vytas7 Would you please also help me on this?

File: docs/user/tutorial.rst
Comment (bssyousefi) [reply docs/user/tutorial.rst:1375]: Sure, I'll look into this.

Comment (bssyousefi) [reply docs/user/tutorial.rst:1375]: @vytas7, I see that a sample usage of query params is being used in the example file for wsgi.
https://github.com/falconry/falcon/blob/65aa7f64d9900380d2246a34d5747782d85d5c8c/examples/things_advanced.py#L158
Are we looking for a more descriptive example?

**Review by CaselIT — APPROVED** [review]

File: docs/user/tutorial.rst
Comment (vytas7) [reply docs/user/tutorial.rst:1375]: Hi again, no, sorry I was not very clear!

No, it has nothing to do with `things_advanced.py`; what I meant was that we are tracking tutorial files in the Git tree as well, we even have some rudimentary tests for them. In this case it looks like you might need to update [`images.py`](https://github.com/falconry/falcon/blob/master/examples/look/look/images.py) with your additions, and potentially even revise how the file develops further in this tutorial.

**Review by vytas7 — CHANGES_REQUESTED** [review]

**Comment by bssyousefi** [comment]
@vytas7 I'm very sorry for the very long delay. I mistakenly thought that all the tests were passed and I didn't got time to get back to the project since I was a little bit busy.
Thanks for you patience.
I will be more available and responsible from now on.

**Comment by vytas7** [comment]
No worries, @bssyousefi :slightly_smiling_face: 
We are now focusing on getting 4.0 out the door, so your PR is not urgent. But it will be nice to have it done when ready.

**Comment by bssyousefi** [comment]
I guess it's ready. I've updated the tests for the example `look`

File: examples/look/look/images.py
Comment (CaselIT) [inline examples/look/look/images.py:20]: let's use media here. What do you think @vytas7 ?

File: examples/look/tests/test_app.py
Comment (CaselIT) [inline examples/look/tests/test_app.py:39]: we can use json() here `response.json()`

**Review by CaselIT — APPROVED** [review]
Nice work, just a couple of minor suggestions

File: examples/look/tests/test_app.py
Comment (bssyousefi) [reply examples/look/tests/test_app.py]: Done

File: examples/look/look/images.py
Comment (vytas7) [reply examples/look/look/images.py:20]: I think it may be a good idea, but in that case we might need to revise the whole tutorial, and either convert to media wholesale (I'm not against it), or illustrate both approaches, but then we need to explain to the reader why we choose one or another.

Comment (CaselIT) [reply examples/look/look/images.py:20]: talked with @vytas7 and we already had in program to use media in the tutorial, but that would be another issue.

so it's fine as is for now!

**Review by CaselIT — APPROVED** [review]

**Comment by bssyousefi** [comment]
Hi,
Is there any issue blocking this PR?

**Comment by vytas7** [comment]
Hi @bssyousefi, and thanks for bringing this up. No, I don't think so, I just need to review it again.

**Review by vytas7 — APPROVED** [review]
Let's merge this as-is, thanks for this improvement @bssyousefi!

It seems that there are a couple of minor inconsistencies, but that could be left as an exercise for the reader. Ultimately, we should move to the approach suggested in #2247, otherwise it is hard to keep track of all the minutiae by hand.

**Pull request merged**
